spacy_experimental/span_finder/tests/test_span_finder.py

```diff
@@ -68,6 +70,40 @@ def test_span_finder_component_span_lengths():
     )
 
 
+def test_span_finder_component_set_annotations_span_lengths():
+
+    test_min_length = 2
+    test_max_length = 3
+
+    nlp = Language()
+    doc = nlp("Me and Jenny goes together like peas and carrots.")
+    span_finder = nlp.add_pipe(
+        "experimental_span_finder",
+        config={"max_length": test_max_length, "min_length": test_min_length},
+    )
+    nlp.initialize()
+    scores = [
+        (1, 0),
+        (0, 0),
+        (1, 1),
+        (0, 0),
+        (0, 0),
+        (0, 0),
+        (1, 1),
+        (0, 0),
+        (0, 1),
+        (0, 1),
+    ]
+    span_finder.set_annotations([doc], scores)
+
+    assert doc.spans[DEFAULT_CANDIDATES_KEY]
+    assert len(
+        doc.spans[DEFAULT_CANDIDATES_KEY][0]
+    ) >= test_min_length and test_max_length >= len(
+        doc.spans[DEFAULT_CANDIDATES_KEY][0]
+    )
```

Comment:
This ordering in the comparisons threw me a bit. It won't let me suggest this directly (not sure why), so something more like:

```python
assert all(test_min_length <= len(span) <= test_max_length for span in doc.spans[DEFAULT_CANDIDATES_KEY])
```